libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/offline/RoutingTileDownloadManager.java

```diff
@@ -0,0 +1,120 @@
+package com.mapbox.services.android.navigation.v5.navigation.offline;
+
+import android.os.AsyncTask;
+import android.os.Environment;
+import android.support.annotation.NonNull;
+
+import com.mapbox.services.android.navigation.v5.utils.DownloadTask;
+
+import java.io.File;
+
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * This class serves to contain the complicated chain of events that must happen to download
+ * offline routing tiles. It creates and maintains a directory structure with the root in the
+ * Offline directory, or wherever someone specifies.
+ */
+public class RoutingTileDownloadManager {
+
+  enum State {
+    NOT_STARTED, WAITING_FOR_RESPONSE, DOWNLOADING_TAR, UNPACKING_TAR, COMPLETED
+  }
+
+  private State state = State.NOT_STARTED; // todo manage errors sent to listener with state info
+  private final File destDirectory;
+  private final File tileDirectory;
+  private String version;
+
+  private RoutingTileDownloadListener listener;
+
+  public RoutingTileDownloadManager() {
+    this(Environment.getExternalStoragePublicDirectory("Offline"));
+  }
+
+  public RoutingTileDownloadManager(File destDirectory) {
+    this.destDirectory = destDirectory;
+    tileDirectory = new File(destDirectory, "tiles");
+  }
+
+  public void setListener(RoutingTileDownloadListener listener) {
+    this.listener = listener;
+  }
+
+  public void startDownloadChain(final OfflineTiles.Builder builder) {
```

Comment:
Does it make sense to pass in the built `OfflineTiles` here instead of the `Builder`?  It looks like we would be able to expose [`MapboxRouteTiles#version()`](https://github.com/mapbox/mapbox-java/blob/master/services-route-tiles/src/main/java/com/mapbox/api/routetiles/v1/MapboxRouteTiles.java#L53) in the `OfflineTiles` class.